models/provenance.py

```diff
@@ -52,7 +51,7 @@ def update(self, session=None):
         if git_hash is None:
             return  # quietly fail if we can't get the git hash
         with SmartSession(session) as session:
-            hash_obj = session.scalars(sa.select(CodeHash).where(CodeHash.hash == git_hash)).first()
+            hash_obj = session.scalars(sa.select(CodeHash).where(CodeHash.id == git_hash)).first()
```

Comment:
Are you sure you also changed CodeHash's id and not only the CodeVersion?

Reply:
Yeah -- I renamed hash to id, and changed it from unique and index to primary key (which makes it unique and indexed).  Since I don't put in the AutoIDMixin here, it gets rid of the old auto-incrementing integer id.